src/S3/S3Client.php

```diff
@@ -623,7 +623,9 @@ private function getLocationConstraintMiddleware()
         $region = $this->getRegion();
         return static function (callable $handler) use ($region) {
             return function (Command $command, $request = null) use ($handler, $region) {
-                if ($command->getName() === 'CreateBucket') {
+                if ($command->getName() === 'CreateBucket'
+                    && empty($command['CreateBucketConfiguration']['LocationConstraint'])
+                ) {
                     if ($region === 'us-east-1') {
```

Comment:
I don't see `us-east-1` listed as a valid LocationConstraint in the S3 docs: http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketPUT.html. You're supposed to omit the LocationConstraint altogether if you want to create a bucket in `us-east-1`. I was saying that the pull request should be updated to also remove the LocationConstraint if the value it's set to is `us-east-1`.